lib/merge_supporters.rb

```diff
@@ -84,9 +101,11 @@ def self.merge_by_id_groups(np_id, arr_of_ids, profile_id)
           acc
         end.merge({'nonprofit_id' => np_id})
 
-        MergeSupporters.selected(data, ids, np_id, profile_id)
+        result = MergeSupporters.selected(data, ids, np_id, profile_id, skip_conflicting_custom_fields)
+        conflicting_custom_fields << result[:json] if result[:status] == :failure
       end
     end
+    conflicting_custom_fields
```

Comment:
This would change the returned value of `.merge_by_id_groups` to change from the `arr_of_ids` to the array of `conflicting_custom_fields`. I don't see why that'd be a problem but make sure to check that nothing actually uses that return value.

Also, since this is not super intuitive given the name of the method, make sure to document the return value for the method in using the [@return YARD tag](https://rubydoc.info/gems/yard/file/docs/Tags.md#return).